prow/cmd/branchprotector/protect.go

```diff
@@ -315,67 +306,95 @@ func (p *protector) UpdateBranch(orgName, repo string, branchName string, branch
+			equalStringSlices(&state.Contexts, &request.Contexts)
+	default:
+		return false
+	}
+}
+
+func equalStringSlices(s1, s2 *[]string) bool {
+	switch {
+	case s1 == s2:
+		return true
+	case s1 != nil && s2 != nil:
+		if len(*s1) != len(*s2) {
+			return false
+		}
+		sort.Strings(*s1)
+		sort.Strings(*s2)
+		for i, v := range *s1 {
+			if v != (*s2)[i] {
 				return false
 			}
-			sort.Strings(*s1)
-			sort.Strings(*s2)
-			for i, v := range *s1 {
-				if v != (*s2)[i] {
-					return false
-				}
-			}
-			return true
-		default:
-			return false
 		}
+		return true
+	default:
+		return false
 	}
+}
 
-	equalStatusChecks := func(sc1, sc2 *github.RequiredStatusChecks) bool {
-		switch {
-		case sc1 == sc2:
-			return true
-		case sc1 != nil && sc2 != nil:
-			return sc1.Strict == sc2.Strict &&
-				equalStringSlices(&sc1.Contexts, &sc2.Contexts)
-		default:
-			return false
-		}
+func equalAdminEnforcement(state github.EnforceAdmins, request *bool) bool {
+	switch {
+	case request == nil:
+		// requesting nothing is the same as requesting this to be off
+		return state.Enabled == false
+	default:
+		return state.Enabled == *request
 	}
+}
 
-	equalRestrictions := func(r1, r2 *github.Restrictions) bool {
-		switch {
-		case r1 == r2:
-			return true
-		case r1 != nil && r2 != nil:
-			return equalStringSlices(r1.Teams, r2.Teams) && equalStringSlices(r1.Users, r2.Users)
-		default:
-			return false
-		}
+func equalRequiredPullRequestReviews(state *github.RequiredPullRequestReviews, request *github.RequiredPullRequestReviewsRequest) bool {
+	switch {
+	case state == nil && request == nil:
+		return true
+	case state != nil && request != nil:
+		return state.DismissStaleReviews == request.DismissStaleReviews &&
+			state.RequireCodeOwnerReviews == request.RequireCodeOwnerReviews &&
+			state.RequiredApprovingReviewCount == request.RequiredApprovingReviewCount &&
+			equalRestrictions(state.DismissalRestrictions, &request.DismissalRestrictions)
+	default:
+		return false
 	}
+}
 
-	equalReviews := func(r1, r2 *github.RequiredPullRequestReviews) bool {
-		switch {
-		case r1 == r2:
-			return true
-		case r1 != nil && r2 != nil:
-			return r1.DismissStaleReviews == r2.DismissStaleReviews &&
-				r1.RequireCodeOwnerReviews == r2.RequireCodeOwnerReviews &&
-				r1.RequiredApprovingReviewCount == r2.RequiredApprovingReviewCount &&
-				equalRestrictions(&r1.DismissalRestrictions, &r2.DismissalRestrictions)
-		default:
-			return false
+func equalRestrictions(state *github.Restrictions, request *github.RestrictionsRequest) bool {
+	switch {
+	case state == nil && request == nil:
+		return true
+	case state == nil && request != nil:
+		return request.Users == nil && request.Teams == nil
+	case state != nil && request != nil:
+		var users []string
+		for _, user := range state.Users {
+			users = append(users, user.Login)
+		}
+		var teams []string
+		for _, team := range state.Teams {
+			teams = append(teams, team.Slug)
```

Comment:
Please leave a comment explaining why the team name in config is compared against the team slug name from the API response instead of the team name.

Reply:
Left one on the `Requirements` struct, will copy here